spot_driver_cpp/src/interfaces/rclcpp_publisher_interface.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver_cpp/interfaces/rclcpp_publisher_interface.hpp>
+
+namespace spot_ros2
+{
+
+RclcppPublisherInterface::RclcppPublisherInterface(const std::shared_ptr<rclcpp::Node>& node)
+: node_{ node }
+{
+}
+
+void RclcppPublisherInterface::createPublishers(const std::vector<ImageSource>& image_sources)
+{
+    image_publishers_.clear();
+    info_publishers_.clear();
+
+    for (const auto& image_source : image_sources)
+    {
+        // Since these topic names do not have a leading `/` character, they will be published within the namespace of the node, which should match the name of the robot.
+        // For example, the topic for the front left RGB camera will ultimately appear as `/MyRobotName/camera/frontleft/image`.
+        const auto topic_name_base = toRosTopic(image_source);
+
+        const auto image_topic_name = topic_name_base + "/image";
+        image_publishers_.try_emplace(image_topic_name, node_->create_publisher<sensor_msgs::msg::Image>(image_topic_name, rclcpp::QoS(1)));
```

Comment:
There should only be a single ImageSource for a given combination of camera type and image type, so there should be exactly one possible entry into this map for each ImageSource.

To do a better job of enforcing this, I changed how the collection of ImageSources is generated and passed around so it's stored in a `std::set` instead of a `std::vector`. This more clearly communicates our expectation that there cannot be duplicate ImageSources.